### PR TITLE
Zab/move blocks ioctl

### PR DESCRIPTION
**Pull request by zabbo**

### CODE REVIEW

File: kmod/src/data.c [hunk 1]
```diff
@@ -1103,6 +1104,240 @@ int scoutfs_data_init_offline_extent(struct inode *inode, u64 size,
...
+
+		ret = scoutfs_dirty_inode_item(from, from_lock) ?:
+		      scoutfs_dirty_inode_item(to, to_lock);
+		if (ret < 0)
+			goto out;
+
+		down_write_two(&from_si->extent_sem, &to_si->extent_sem);
+
+		/* arbitrarily limit the number of extents per trans hold */
+		for (i = 0; i < 16; i++) {
```
Comment (agrover) [inline kmod/src/data.c:1241]: #define EXTENTS_PER_TRANS_HOLD 16 ?

File: kmod/src/ioctl.c [hunk 2]
```diff
@@ -937,6 +938,55 @@ static long scoutfs_ioc_alloc_detail(struct file *file, unsigned long arg)
...
+		goto out;
+	}
+
+	if (from->i_sb != sb) {
+		ret = -EXDEV;
+		goto out;
+	}
+
+	/* is it enough to only get one file? */
+	ret = mnt_want_write_file(file);
```
Comment (agrover) [inline kmod/src/ioctl.c:977]: dunno, looks like you could call `mnt_clone_write()` here if you wanted (and then drop_write() twice) but what would be the point?

File: kmod/src/data.c (hunk 1 above)
Comment (zabbo) [reply kmod/src/data.c:1241]: Sold!

File: kmod/src/ioctl.c (hunk 2 above)
Comment (zabbo) [reply kmod/src/ioctl.c:977]: Huh, I hadn't noticed that call.  But now that I've dug in, I don't think we need more than one.  I was worried that the _file() variant actually did something to the file, but it doesn't, it just uses it to get at the file's mount point.  So one call is enough 'cause we specifically require that the two files be on one mount.  I'll remove the comment.

**Pull request merged**